gp_predict.f95

```diff
@@ -799,8 +799,9 @@ subroutine gpSparse_fit(this, from, task_manager, condition_number_norm, error)
 
       allocate(alpha(n_globalSparseX))
       if (task_manager%active) then
-         nlrows = (task_manager%unified_workload / n_globalSparseX) * n_globalSparseX
-         if (nlrows < task_manager%unified_workload) nlrows = nlrows + n_globalSparseX
+         blocksize = ifel(task_manager%idata(1) > 0, task_manager%idata(1), n_globalSparseX)
```

Comment:
try the built-in ternary?

Reply:
Works like a charm. Thanks for the hint.